src/dune_engine/build_system.ml

```diff
@@ -1192,8 +1194,8 @@ end = struct
     let static_deps (type a) (t : a t) = Build.static_deps (build t)
 
     let evaluate_and_discover_dynamic_deps_unmemoized t =
-      let+ () = build_deps (static_deps t).rule_deps in
-      Build.exec (build t)
+      let* () = build_deps (static_deps t).rule_deps in
+      Build_exec.exec (build t)
```

Comment:
It seems to me that we should now be able to just call `exec` directly, using the `Build` case (although I'm not entirely sure it's a good idea).

Reply:
In a way, we already expose `Build.go_deps t`: it's `Build.exec (Build (Pure t))` if I'm not mistaken.

Anyway, I'm fine if we keep the code as is but it would be nice to add a comment pointing out that there is some code duplication between these two lines and the `Build` case in `build.ml`. (I'll add this comment via suggestions.)